JSDemos/Demos/Charts/PieWithCustomStyles/Angular/app/app.component.ts

```diff
@@ -0,0 +1,146 @@
+import { NgModule, Component, enableProdMode } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { DxPieChartModule } from 'devextreme-angular';
+import { registerGradient, registerPattern } from 'devextreme/common/charts';
+import { Data, Service } from './app.service';
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app/app.component.html',
+  styleUrls: ['app/app.component.css'],
+  providers: [Service],
+})
+
+export class AppComponent {
+  data: Data[];
+
+  imagePatternSize = 12;
+
+  shapePatternSize = 6;
+
+  constructor(service: Service) {
+    this.data = service.getData();
+  }
+
+  customizePoint = (point) => {
+    const color = point.series.getPointsByArg(point.argument)[0].getColor();
+    const pointSettings = { color };
+    if (point.argument === 'Stroke Pattern') {
+      pointSettings.color = { fillId: this.strokePattern(color) };
+    }
+    if (point.argument === 'Square Pattern') {
+      pointSettings.color = { fillId: this.squarePattern(color) };
+    }
+    if (point.argument === 'Linear Gradient') {
+      pointSettings.color = { fillId: this.linearGradient(color) };
+    }
+    if (point.argument === 'Radial Gradient') {
+      pointSettings.color = { fillId: this.radialGradient(color) };
+    }
+    if (point.argument === 'Pattern with image') {
+      pointSettings.color = { fillId: this.patternImage(color) };
+    }
+
+    return pointSettings;
+  };
+
+  customizeText = (info) => info.argument;
+
+  hexToRgb = (hex: string, opacity = 1) => {
+    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(result[3], 16)}, ${opacity})`;
+  };
+
+  gradient = (type: string, color1: string, color2: string) => registerGradient(type, {
+    colors: [{
+      offset: '20%',
+      color: color1,
+    }, {
+      offset: '90%',
+      color: color2,
+    }],
+  });
+
+  linearGradient = (color: string) => this.gradient('linear', color, this.hexToRgb(color, 0.5));
```

Comment:
It's better to name function starting a verb.
Maybe `getLinearGradient` or `createLinearGradient`?